R/to_relational_data.R

```diff
@@ -120,9 +120,3 @@ to_relational_data.list <- function(data) {
 to_relational_data.MultiAssayExperiment <- function(data) { # nolint
   teal_data(dataset("MAE", data))
 }
-
-#' @keywords internal
-#' @export
-to_relational_data.TealData <- function(data) { # nolint
-  data
-}
```

Comment:
why?

This potentially could be quite dangerous - you would have to track down and analyse all the usage of `to_relational_data()`. From what I recall its used in apply through a list of elements of unknown class. If it happens that the element is `TealData` then after this change we will have an error.

Reply:
This function is used in `teal::init` I removed above method and decided to make [if condition instead](https://github.com/insightsengineering/teal/blob/e8c9d8623840edb71fbb38e6ed31052f4d7c961b/R/init.R#L118)

If I didn't make this if, I would have to introduce `to_relational.teal_data` also. `to_relational` will be removed after internal release together with all `TealData` related stuff